### PR TITLE
Removed Metrics from the Event Loop

### PR DESCRIPTION
**Pull request by momosh-ethernal**

### CODE REVIEW

File: client/src/main.rs [hunk 1]
```diff
@@ -98,7 +98,13 @@ async fn run(
...
 	.await?;
 
-	spawn_in_span(shutdown.with_cancel(p2p_event_loop.run(ot_metrics.clone())));
+	spawn_in_span(shutdown.with_cancel(p2p_event_loop.run()));
+
+	let metrics_clone = ot_metrics.clone();
+	tokio::spawn(async move {
+		let stream = BroadcastStream::new(event_receiver.resubscribe());
+		metrics_clone.handle_event_stream(stream).await
+	});
```
Comment (aterentic-ethernal) [inline client/src/main.rs:107]: We should run event loop after the call to resubscribe to avoid skipping an event.

File: crawler/src/main.rs [hunk 2]
```diff
@@ -100,7 +101,13 @@ async fn run(config: Config, db: DB, shutdown: Controller<String>) -> Result<()>
 	)
 	.await?;
 
-	spawn_in_span(shutdown.with_cancel(p2p_event_loop.run(ot_metrics.clone())));
+	spawn_in_span(shutdown.with_cancel(p2p_event_loop.run()));
```
Comment (aterentic-ethernal) [inline crawler/src/main.rs:104]: Same thing for this run, lets move it after resubscribe.

File: fat/src/main.rs [hunk 3]
```diff
@@ -100,7 +101,13 @@ async fn run(config: Config, db: DB, shutdown: Controller<String>) -> Result<()>
 	)
 	.await?;
 
-	spawn_in_span(shutdown.with_cancel(p2p_event_loop.run(ot_metrics.clone())));
+	spawn_in_span(shutdown.with_cancel(p2p_event_loop.run()));
```
Comment (aterentic-ethernal) [inline fat/src/main.rs:104]: Same thing for this run, lets move it after resubscribe.

File: core/src/network/p2p/event_loop.rs [hunk 4]
```diff
@@ -346,45 +347,21 @@ impl EventLoop {
...
-										stats.clone(),
-										true,
-										metrics,
-									)
-									.await;
-
-									_ = self
-										.event_sender
-										.send(OutputEvent::PutRecordQuorumFailed { key, stats });
+									self.handle_put_result(key.clone(), stats.clone(), true)
```
[the remark this answers is not in the document]
Comment (momosh-ethernal) [reply core/src/network/p2p/event_loop.rs:350]: Yes, of course. This functionality depends on the `BlockStat` struct, which I was hoping to decouple and move in the next iteration.

**Review by sh3ll3x3c — APPROVED** [review]

**Review by sh3ll3x3c — APPROVED** [review]

**Review by sh3ll3x3c — APPROVED** [review]

**Review by aterentic-ethernal — APPROVED** [review]

**Pull request merged**